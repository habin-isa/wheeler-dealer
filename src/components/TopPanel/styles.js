import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-evenly;
  align-items: center;
  position: relative;
  padding-bottom: 10px;
`;

export const Results = styled.div`
`;

export const HeartContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Heart = styled.img`
  width: 40px;
  padding: 20px;
`;

export const Button = styled.button`
  width: 150px;
  height: 40px;
  border-color: red;
  background: red;
  color: white;
  font-weight: bold;
  outline: none;
  cursor: pointer;
`;

export const Notification = styled.div`
  position: absolute;
  cursor: pointer;
  font-size: 14px;
`;