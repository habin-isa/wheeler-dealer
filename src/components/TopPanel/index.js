import React, { useState } from 'react';
import { func, number, bool, array } from 'prop-types';
import * as S from './styles';
import Notification from '../Notification';

import likeIcon from '../../assets/full-like.svg'

const TopPanel = ({ productCount, likeCount, handleShowClick, hideSoldItems, deleteLike, likedList }) => {
  const [showSoldButton, setShowSoldButton] = useState(false);
  
  const handleClick = () => {
    setShowSoldButton(!showSoldButton);
    if (hideSoldItems === false) {
      handleShowClick(false);
    } else {
      handleShowClick(true);
    }
	};

  return (
    <S.Wrapper>
      <S.Results>{productCount} Results</S.Results>
      {
        !showSoldButton ? (
          <S.Button onClick={handleClick}>Hide sold items</S.Button>
        ) : (
          <S.Button onClick={handleClick}>Show all items</S.Button>
        )
      }
      <S.NotificationWrapper>
        <S.HeartContainer>
          <S.Heart src={likeIcon} /> {likeCount}
        </S.HeartContainer>
        <Notification likedList={likedList} deleteLike={deleteLike} />
      </S.NotificationWrapper>
    </S.Wrapper>
  )
};

TopPanel.propTypes = {
  productCount: number,
  likeCount: number,
  handleShowClick: func,
  hideSoldItems: bool,
  deleteLike: func,
  likedList: array
}

export default TopPanel;