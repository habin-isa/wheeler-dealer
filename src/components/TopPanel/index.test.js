
import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import TopPanel from '.';

describe('Top Panel', () => {

  const props = {
    productCount: 5,
    likeCount: 3,
  };

  const likedList = [
    'Man City shirt',
    'Buberry slippers',
    'Rapha top'
  ];

  it('should render', () => {
    const { container } = render(<TopPanel {...props} likedList={likedList} />);
    expect(container.firstChild).toMatchSnapshot();
  });

   it('renders with correct count data', () => {
      shallow(<TopPanel product={props} likedList={likedList} />);
    });

});