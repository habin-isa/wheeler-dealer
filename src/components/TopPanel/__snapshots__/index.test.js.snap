// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Panel should render 1`] = `
<div
  class="sc-AxjAm fMYzc"
>
  <div
    class="sc-AxirZ dqVvws"
  >
    5
     Results
  </div>
  <button
    class="sc-AxhUy iJJDLQ"
  >
    Hide sold items
  </button>
  <div
    class="sc-AxgMl ecWowd"
  >
    <div
      class="sc-AxiKw hQHhPG"
    >
      <img
        class="sc-AxhCb dKRPPF"
        src="full-like.svg"
      />
       
      3
    </div>
    <div
      class="sc-AxheI hBERmv"
    >
      <li
        class="sc-Axmtr eKrwkX"
      >
        ⤬ New like: 
        Man City shirt
      </li>
      <li
        class="sc-Axmtr eKrwkX"
      >
        ⤬ New like: 
        Buberry slippers
      </li>
      <li
        class="sc-Axmtr eKrwkX"
      >
        ⤬ New like: 
        Rapha top
      </li>
    </div>
  </div>
</div>
`;
